components/blitz/src/omero/gateway/facility/AdminFacility.java

```diff
@@ -427,6 +431,78 @@ public void removeAdminPrivileges(SecurityContext ctx,
     }
 
     /**
+     * Get all available admin privileges
+     * 
+     * @param ctx
+     *            The security context.
+     * @return See above.
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in.
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public Collection<String> getAvailableAdminPrivileges(SecurityContext ctx)
+            throws DSOutOfServiceException, DSAccessException {
+        if (adminPrivileges == null) {
+            try {
+                List<IObject> tmp = gateway.getTypesService(ctx)
+                        .allEnumerations("AdminPrivilege");
+                List<String> tmp2 = new ArrayList<String>(tmp.size());
```

Comment:
Maybe can just use `Utils.fromEnum`?

Reply:
Sorry, of course, makes it much simpler. 👍